src/Components/Modals/EditAdditionalRepositoriesModal.js

```diff
@@ -1,10 +1,48 @@
-import React from 'react';
+import React, { useState } from 'react';
 import propTypes from 'prop-types';
-import { Modal, ModalVariant, Button } from '@patternfly/react-core';
+import { Modal, ModalVariant } from '@patternfly/react-core';
 import EditAdditionalRepositoriesTable from '../EditAdditionalRepositoriesTable/EditAdditionalRepositoriesTable';
+import { useQueryClient } from 'react-query';
+import useAddAdditionalRepositories from '../../hooks/useAddAdditionalRepositories';
+import useNotifications from '../../hooks/useNotifications';
 
 const EditAdditionalRepositoriesModal = (props) => {
-  const { handleModalToggle, isOpen, repositories, isLoading, error } = props;
+  const { keyName, handleModalToggle, isOpen, repositories } = props;
+  const queryClient = useQueryClient();
+
+  const [additionalRepos, setAdditionalRepos] = useState([]);
+
+  const [created, setCreated] = React.useState(false);
+  const [error, setError] = React.useState(false);
+  const { addSuccessNotification, addErrorNotification } = useNotifications();
+  const { mutate, isLoading } = useAddAdditionalRepositories();
+  const submitForm = () => {
+    mutate(
+      { additionalRepos, keyName },
+      {
+        onSuccess: () => {
+          setError(false);
+          setCreated(true);
+          queryClient.invalidateQueries('activation_keys');
+          queryClient.resetQueries(`activation_key_${keyName}`);
+          queryClient.resetQueries(
+            `activation_key_${keyName}_available_repositories`
+          );
```

Comment:
Do we need to mess with all of these queries when an additional repo is edited? I wouldn't think you need to refetch all the activation keys.